plugins/teststeps/s3fileupload/s3FileUpload.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package s3fileupload
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/facebookincubator/contest/pkg/event"
+	"github.com/facebookincubator/contest/pkg/event/testevent"
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/test"
+	"github.com/facebookincubator/contest/pkg/xcontext"
+	"github.com/facebookincubator/contest/plugins/teststeps"
+)
+
+// Name is the name used to look this plugin up.
+var Name = "s3FileUpload"
+
+// Event names for this plugin.
+const (
+	EventURL = event.Name("URL")
+)
+
+// Events defines the events that a TestStep is allow to emit
+var Events = []event.Name{
+	EventURL,
+}
+
+// FileUpload is used to retrieve all the parameter, the plugin needs.
+type FileUpload struct {
+	localpath     *test.Param // Path to file that shall be uploaded
```

Comment:
field names (unlike files) should use camelCase, hence `localPath`, `fileName`, etc.